docker/loadbalancer.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/klog/klogr"
+	infrav1 "sigs.k8s.io/cluster-api-provider-docker/api/v1alpha2"
+	"sigs.k8s.io/cluster-api-provider-docker/third_party/forked/loadbalancer"
+	clusterapiv1alpha2 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/kind/pkg/cluster/constants"
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+)
+
+// LoadBalancer implement a service for managing the docker containers hosting a loadbalancer.
+type LoadBalancer struct {
+	logr.Logger
+	Cluster       *clusterapiv1alpha2.Cluster
+	DockerCluster *infrav1.DockerCluster
+}
+
+// NewLoadBalancer returns a new LoadBalancer service for the given Cluster/DockerCluster pair.
+func NewLoadBalancer(cluster *clusterapiv1alpha2.Cluster, dockerCluster *infrav1.DockerCluster, logger logr.Logger) (*LoadBalancer, error) {
```

Comment:
This doesn't need the whole cluster, it should just be the load balancer name. This object also does not use the `dockerCluster` at all.

This should be `NewLoadBalancer(name string, log logr.Logger) (*LoadBalancer, error)`